mutation_partition_view.cc

```diff
@@ -286,6 +286,7 @@ mutation_partition_view::accept_ordered_cookie<reverse>::rts_crs_iterators::rts_
 
 template <bool is_preemptible, consume_in_reverse reverse>
 mutation_partition_view::accept_ordered_result<reverse> mutation_partition_view::do_accept_ordered(const schema& schema, mutation_partition_view_virtual_visitor& visitor, accept_ordered_cookie<reverse> cookie) const {
+    static_assert(reverse != consume_in_reverse::legacy_half_reverse);
```

Comment:
The patches that get rid of legacy_half_reverse should come first, potentially even merged separately, if this PR gets delayed for some reason.
This will simplify the unit tests that are added early on, as they are bogus for legacy_half_revrerse

Reply:
I hoped to avoid that (it seems like a fair bit of useless work) but if you deem that necessary I can do that.

Reply:
yes, please, we try to get the patches in the best shape we can